haystack/utils/hf.py

```diff
@@ -21,6 +21,28 @@
 logger = logging.getLogger(__name__)
 
 
+class HFGenerationAPIType(Enum):
+    """
+    API type to use for Hugging Face API Generators.
+    """
+
+    TEXT_GENERATION_INFERENCE = "text_generation_inference"
+    INFERENCE_ENDPOINTS = "inference_endpoints"
+    SERVERLESS_INFERENCE_API = "serverless_inference_api"
```

Comment:
```suggestion
	#: HF [Text Generation Inference](<url>) (TGI).
    TEXT_GENERATION_INFERENCE = "text_generation_inference"

    #: ...
    INFERENCE_ENDPOINTS = "inference_endpoints"
    
    #: ...
    SERVERLESS_INFERENCE_API = "serverless_inference_api"
```
Let's add docstring for each variant that links to the corresponding HF docs page.

Reply:
done!